### ISSUE TITLE
[QST]: Early stopping or checkpoints while training retrieval model (Two tower model)

### ISSUE THREAD

**Issue by Abhinavb08** [NVIDIA-Merlin/models#919]
# ❓ Questions & Help

## Details

Hello Merlin Team,

Please point me to the documentation or code for doing early stopping for two tower model training. I am using the latest TwotowermodelV2 which is shown in the example notebook.
Example notebook link: https://github.com/NVIDIA-Merlin/models/blob/main/examples/05-Retrieval-Model.ipynb

**Comment by Abhinavb08** [comment]
@rnyak  Please let me know in case you want more information or context.

**Comment by rnyak** [comment]
@Abhinavb08 did you try to use early stopping as it is done in native tf.keras using callbacks like explained in this [doc](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping)?

once you define `callback = tf.keras.callbacks.EarlyStopping(monitor='loss', patience=3)` then you should be add it in the model.fit() as in this example 
```
model.fit(train, validation_data=valid, batch_size=4096, epochs=2,  callbacks=[callback] )
```

let us know if this does not work for you or you hit any issue.. thanks.

**Comment by Abhinavb08** [comment]
This was helpful. Thanks for your response.

**Comment by Abhinavb08** [comment]
It worked.
